HEN_HOUSE/user_codes/egs_kerma/egs_kerma.cpp

```diff
@@ -239,19 +239,20 @@ class APP_EXPORT EGS_KermaApplication : public EGS_AdvancedApplication {
 
                     }
                 }
-                /* Mark photon inside cavity */
-                latch = 1;
+                /* Mark photon inside cavity <= why did I need this?*/
+                //latch = 1;
             }
             /* Mark photons in exclusion regions */
-            else if (is_excluded[ig][ir]) {
-                latch = -1;
+            else if (is_excluded[ig][ir] && latch >= 0 ) {
+                latch *= -1;
+                the_stack->latch[np] = latch;
```

Comment:
@mainegra, I'm not totally sure how latch is being used.  From the logic, it looks to me like if latch=0--true unless egs++ DBS is being used or latch has been set in a phase space or BEAMnrc source--then we can't mark the photon as having been in an exclusion region (i.e. latch *=-1 = 0).  Does this amount to a necessary restriction on when exclusion regions can be used or am I missing something?

Reply:
The current assumption is that if a particle enters an exclusion zone, it is marked with a negative latch, and it does not contribute to kerma. One could save time and simply kill it, I guess. According to this, there wouldn't be a conflict.

Reply:
But what if latch=0 going into the simulation?  Can the photon be marked on entering the exclusion zone then?